packages/block-editor/src/components/use-block-drop-zone/index.js

```diff
@@ -174,12 +237,10 @@ export default function useBlockDropZone( {
 		onDragLeave() {
 			throttled.cancel();
 			hideInsertionPoint();
-			setTargetBlockIndex( null );
```

Comment:
I'm actually not sure why we needed to set it to `null` here as it'll get updated on drag over anyway?

This is making the hook returned by `useOnBlockDrop` receive the wrong index sometimes.

Reply:
Note that there is some flickering in Safari when dragging in between two blocks:

https://user-images.githubusercontent.com/7753001/193735124-9d689e41-f8fa-44c8-932f-59aab71f8112.mp4

It's because Safari will always [return `null` for `event.relatedTarget` in the `dragleave` events](https://bugs.webkit.org/show_bug.cgi?id=66547). I think there are ways to fix it but probably deserves another PR.